src/Components/test/E2ETest/Tests/InteropTest.cs

```diff
@@ -114,9 +114,9 @@ public void CanInvokeDotNetMethods()
             ["result8"] = @"[{""id"":7,""isValid"":false,""data"":{""source"":""Some random text with at least 7 characters"",""start"":7,""length"":7}},7,123,28,56,7.25,[0.5,1.5,2.5,3.5,4.5,5.5,6.5]]",
             ["result9"] = @"[{""id"":8,""isValid"":true,""data"":{""source"":""Some random text with at least 8 characters"",""start"":8,""length"":8}},8,123,32,64,8.25,[0.5,1.5,2.5,3.5,4.5,5.5,6.5,7.5],{""source"":""Some random text with at least 7 characters"",""start"":9,""length"":9}]",
             ["roundTripJSObjectReference"] = @"""successful""",
-            ["invokeDisposedJSObjectReferenceException"] = @"""JS object instance with ID",
-            ["ThrowException"] = @"""System.InvalidOperationException: Threw an exception!",
-            ["ExceptionFromSyncMethod"] = "Function threw an exception!",
+            ["invokeDisposedJSObjectReferenceException"] = @"""Error: JS object instance with ID",
+            ["ThrowException"] = @"""Threw an exception!",
```

Comment:
Is the exception type name still exposed on the JS side somehow? Seems like that would be valuable for debugging.

Reply:
Unfortunately, it looks like that information is lost now. The JavaScript errors that get created from exceptions thrown in `[JSExport]` methods have a `message` property equivalent to the C# exception's `Message` property, which excludes the exception type and stack trace. We _could_ fix this by catching exceptions on the .NET side and wrapping them in a new `Exception` to simulate the old behavior, like:

```csharp
try
{
    return DotNetDispatcher.Invoke(Instance, callInfo, argsJson);
}
catch (Exception ex)
{
    throw new Exception(ex.ToString(), ex);
}
```

Or I wonder if there's a way to address this on the runtime side, perhaps by updating [`ManagedError`](https://github.com/dotnet/runtime/blob/dc6ad37d410931edb6b46203fb73d6503ed29dc8/src/mono/wasm/runtime/marshal.ts#L319) to include information about the type of exception that got thrown?

Reply:
They are just a proxy and could also cross the boundary back to managed. 
Collecting stack trace is very very expensive and we do it when you call `toString()` on it  (I think).

Reply:
> Collecting stack trace is very very expensive and we do it when you call toString() on it (I think)

@pavelsavara Does that mean we could change the JS-side code to call `throw new Error(exceptionProxy.toString())`, and thereby get back the information we had before? Or is there some other mechanism you'd recommend?

Reply:
no, please don't do that. It would be slow as you would pay the cost of stack trace creation also for exceptions which would be caught and handled silently.

We override the implementation of `ManagedError.stack` property to provide it on demand.

https://github.com/dotnet/runtime/blob/8d938ec1d18de3211040bccca1ea73f2574dc87a/src/mono/wasm/runtime/marshal.ts#L339

I think `error.toString()` and `console.log(error)` would both include `stack` property (at least on chrome). I'm open to suggestions how to improve it.

Proxy in the other direction is also lazy

https://github.com/dotnet/runtime/blob/8d938ec1d18de3211040bccca1ea73f2574dc87a/src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSException.cs#L40